src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -70,13 +72,40 @@ case object UnknownPeriod extends Period {
 }
 
 case class PeriodSum(periods: Set[Period], modifier: Modifier) extends Period {
-  override def addTo(temporal: Temporal): Temporal = ???
 
-  override def get(unit: TemporalUnit): Long = ???
+  var map = scala.collection.mutable.Map[TemporalUnit,Long]()
 
-  override def subtractFrom(temporal: Temporal): Temporal = ???
+  for ( period <- periods; unit <- period.getUnits.asScala )
+    map.get(unit) match {
+      case Some(value) => map.update( unit, value + period.get(unit))
+      case None => map += ( unit -> period.get(unit))
+    }
 
-  override def getUnits: util.List[TemporalUnit] = ???
+  val list = map.keys.toList.sortBy(_.getDuration()).reverse.asJava
+
+  override def addTo(temporal: Temporal): Temporal = {
+    var current = temporal
+
+    for ((u,n) <- map)
+      current = current.plus(n, u)
+
+    current
+  }
+
+  override def get(unit: TemporalUnit): Long =
+    map.getOrElse( unit, throw new UnsupportedTemporalTypeException(""))
```

Comment:
Either make the message more informative than "", or pass null. https://docs.oracle.com/javase/8/docs/api/java/time/temporal/UnsupportedTemporalTypeException.html#UnsupportedTemporalTypeException-java.lang.String-